### ISSUE TITLE
Unable to install cri-dockerd on EKS Amazon Linux 2

### ISSUE THREAD

**Issue by saurabhheda1990** [Mirantis/cri-dockerd#204]
We are trying to use `cri-dockerd` on our EKS clusters nodes while installing we are getting the below error:

```
# yum install -y https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.3/cri-dockerd-0.3.3-3.el7.x86_64.rpm
Loaded plugins: priorities, update-motd, versionlock
cri-dockerd-0.3.3-3.el7.x86_64.rpm                                                                                                                                                                                                                       | 9.2 MB  00:00:00
Examining /var/tmp/yum-root-jk2qwT/cri-dockerd-0.3.3-3.el7.x86_64.rpm: 3:cri-dockerd-0.3.3-3.el7.x86_64
Marking /var/tmp/yum-root-jk2qwT/cri-dockerd-0.3.3-3.el7.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package cri-dockerd.x86_64 3:0.3.3-3.el7 will be installed
--> Processing Dependency: container-selinux >= 2:2.74 for package: 3:cri-dockerd-0.3.3-3.el7.x86_64
amzn2-core                                                                                                                                                                                                                                               | 3.7 kB  00:00:00
--> Processing Dependency: containerd.io >= 1.2.2-3 for package: 3:cri-dockerd-0.3.3-3.el7.x86_64
--> Processing Dependency: libcgroup for package: 3:cri-dockerd-0.3.3-3.el7.x86_64
--> Running transaction check
---> Package cri-dockerd.x86_64 3:0.3.3-3.el7 will be installed
--> Processing Dependency: container-selinux >= 2:2.74 for package: 3:cri-dockerd-0.3.3-3.el7.x86_64
--> Processing Dependency: containerd.io >= 1.2.2-3 for package: 3:cri-dockerd-0.3.3-3.el7.x86_64
---> Package libcgroup.x86_64 0:0.41-21.amzn2 will be installed
--> Finished Dependency Resolution
Error: Package: 3:cri-dockerd-0.3.3-3.el7.x86_64 (/cri-dockerd-0.3.3-3.el7.x86_64)
           **Requires: containerd.io >= 1.2.2-3**
Error: Package: 3:cri-dockerd-0.3.3-3.el7.x86_64 (/cri-dockerd-0.3.3-3.el7.x86_64)
           **Requires: container-selinux >= 2:2.74**
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```
It looks like it is failing due to dependency packages, however, we already have `contained` but not `containerd.io`

```
# yum list | grep containerd
containerd.x86_64                        1.6.19-1.amzn2.0.1           @amzn2extra-docker
containerd-stress.x86_64                 1.6.19-1.amzn2.0.1           amzn2extra-docker
```

```
# cat /etc/*release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
Amazon Linux release 2 (Karoo)
```

Is there any way I can get around this error?

**Comment by afbjorklund** [comment]
You can install `containerd.io` for CentOS 7, from Docker

**Comment by saurabhheda1990** [comment]
Yeah, I tried that but `containerd.io` also requires `container-selinux`

```
Error: Package: containerd.io-1.6.21-3.1.el7.x86_64 (docker-ce-stable)
           Requires: container-selinux >= 2:2.74
```

And `container-selinux` requires lot of other packages

```
# yum install -y http://mirror.centos.org/centos/7/extras/x86_64/Packages/container-selinux-2.107-1.el7_6.noarch.rpm
Loaded plugins: priorities, update-motd, versionlock
container-selinux-2.107-1.el7_6.noarch.rpm                                                                                                                                                     |  39 kB  00:00:00
Examining /var/tmp/yum-root-rq618U/container-selinux-2.107-1.el7_6.noarch.rpm: 2:container-selinux-2.107-1.el7_6.noarch
Marking /var/tmp/yum-root-rq618U/container-selinux-2.107-1.el7_6.noarch.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package container-selinux.noarch 2:2.107-1.el7_6 will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: selinux-policy-base >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: policycoreutils-python for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Running transaction check
---> Package container-selinux.noarch 2:2.107-1.el7_6 will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: selinux-policy-base >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
---> Package policycoreutils-python.x86_64 0:2.5-22.amzn2 will be installed
--> Processing Dependency: setools-libs >= 3.3.8-2 for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libsemanage-python >= 2.5-9 for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: audit-libs-python >= 2.1.3-4 for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: python-IPy for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libselinux-python for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.4)(64bit) for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.2)(64bit) for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libcgroup for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libapol.so.4(VERS_4.0)(64bit) for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: checkpolicy for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libqpol.so.1()(64bit) for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Processing Dependency: libapol.so.4()(64bit) for package: policycoreutils-python-2.5-22.amzn2.x86_64
--> Running transaction check
---> Package audit-libs-python.x86_64 0:2.8.1-3.amzn2.1 will be installed
---> Package checkpolicy.x86_64 0:2.5-6.amzn2 will be installed
---> Package container-selinux.noarch 2:2.107-1.el7_6 will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: selinux-policy-base >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-216.el7 for package: 2:container-selinux-2.107-1.el7_6.noarch
---> Package libcgroup.x86_64 0:0.41-21.amzn2 will be installed
---> Package libselinux-python.x86_64 0:2.5-12.amzn2.0.2 will be installed
---> Package libsemanage-python.x86_64 0:2.5-11.amzn2 will be installed
---> Package python-IPy.noarch 0:0.75-6.amzn2.0.1 will be installed
---> Package setools-libs.x86_64 0:3.3.8-2.amzn2.0.2 will be installed
--> Finished Dependency Resolution
Error: Package: 2:container-selinux-2.107-1.el7_6.noarch (/container-selinux-2.107-1.el7_6.noarch)
           Requires: selinux-policy-targeted >= 3.13.1-216.el7
           Installed: selinux-policy-targeted-3.13.1-192.amzn2.6.8.noarch (installed)
               selinux-policy-targeted = 3.13.1-192.amzn2.6.8
           Available: selinux-policy-targeted-3.13.1-166.amzn2.5.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-166.amzn2.5
           Available: selinux-policy-targeted-3.13.1-166.amzn2.9.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-166.amzn2.9
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-192.amzn2.6
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.1.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-192.amzn2.6.1
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.2.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-192.amzn2.6.2
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.3.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-192.amzn2.6.3
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.5.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-192.amzn2.6.5
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.7.noarch (amzn2-core)
               selinux-policy-targeted = 3.13.1-192.amzn2.6.7
Error: Package: 2:container-selinux-2.107-1.el7_6.noarch (/container-selinux-2.107-1.el7_6.noarch)
           Requires: selinux-policy-base >= 3.13.1-216.el7
           Installed: selinux-policy-targeted-3.13.1-192.amzn2.6.8.noarch (installed)
               selinux-policy-base = 3.13.1-192.amzn2.6.8
           Available: selinux-policy-minimum-3.13.1-166.amzn2.5.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-166.amzn2.5
           Available: selinux-policy-minimum-3.13.1-166.amzn2.9.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-166.amzn2.9
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.1.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.1
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.2.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.2
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.3.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.3
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.5.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.5
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.7.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.7
           Available: selinux-policy-minimum-3.13.1-192.amzn2.6.8.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.8
           Available: selinux-policy-mls-3.13.1-166.amzn2.5.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-166.amzn2.5
           Available: selinux-policy-mls-3.13.1-166.amzn2.9.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-166.amzn2.9
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.1.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.1
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.2.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.2
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.3.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.3
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.5.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.5
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.7.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.7
           Available: selinux-policy-mls-3.13.1-192.amzn2.6.8.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.8
           Available: selinux-policy-targeted-3.13.1-166.amzn2.5.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-166.amzn2.5
           Available: selinux-policy-targeted-3.13.1-166.amzn2.9.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-166.amzn2.9
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.1.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.1
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.2.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.2
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.3.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.3
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.5.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.5
           Available: selinux-policy-targeted-3.13.1-192.amzn2.6.7.noarch (amzn2-core)
               selinux-policy-base = 3.13.1-192.amzn2.6.7
Error: Package: 2:container-selinux-2.107-1.el7_6.noarch (/container-selinux-2.107-1.el7_6.noarch)
           Requires: selinux-policy >= 3.13.1-216.el7
           Installed: selinux-policy-3.13.1-192.amzn2.6.8.noarch (installed)
               selinux-policy = 3.13.1-192.amzn2.6.8
           Available: selinux-policy-3.13.1-166.amzn2.5.noarch (amzn2-core)
               selinux-policy = 3.13.1-166.amzn2.5
           Available: selinux-policy-3.13.1-166.amzn2.9.noarch (amzn2-core)
               selinux-policy = 3.13.1-166.amzn2.9
           Available: selinux-policy-3.13.1-192.amzn2.6.noarch (amzn2-core)
               selinux-policy = 3.13.1-192.amzn2.6
           Available: selinux-policy-3.13.1-192.amzn2.6.1.noarch (amzn2-core)
               selinux-policy = 3.13.1-192.amzn2.6.1
           Available: selinux-policy-3.13.1-192.amzn2.6.2.noarch (amzn2-core)
               selinux-policy = 3.13.1-192.amzn2.6.2
           Available: selinux-policy-3.13.1-192.amzn2.6.3.noarch (amzn2-core)
               selinux-policy = 3.13.1-192.amzn2.6.3
           Available: selinux-policy-3.13.1-192.amzn2.6.5.noarch (amzn2-core)
               selinux-policy = 3.13.1-192.amzn2.6.5
           Available: selinux-policy-3.13.1-192.amzn2.6.7.noarch (amzn2-core)
               selinux-policy = 3.13.1-192.amzn2.6.7
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

**Comment by neersighted** [comment]
In general, the EL packages here are designed for the Docker CE repositories + a more 'upstream' set of Enterprise Linux packages. It's certainly possible to adapt the packages to work with version numbers provided by AL2, but as none of the current maintainers use AL2, that is a bit fraught.

If you're interested in doing the legwork and expect to keep using these packages going forward, contributions are welcome!

**Comment by nwneisen** [comment]
@saurabhheda1990 I'm closing this issue as it is something that currently out of scope for the project. If you decide you would like to make a contribution, please open an issue to outline your plans with a followup PR. Thanks!
